pins/drivers.py

```diff
@@ -149,39 +145,22 @@ def save_data(
         final_name = f"{fname}{suffix}"
 
     if pin_type == "csv":
-        _assert_is_pandas_df(obj, file_type=type)
-
-        obj.to_csv(final_name, index=False)
-
+        adaptor.write_csv(final_name)
     elif pin_type == "arrow":
         # NOTE: R pins accepts the type arrow, and saves it as feather.
         #       we allow reading this type, but raise an error for writing.
-        _assert_is_pandas_df(obj, file_type=type)
-
-        obj.to_feather(final_name)
-
+        adaptor.write_feather(final_name)
     elif pin_type == "feather":
-        _assert_is_pandas_df(obj, file_type=type)
-
-        raise NotImplementedError(
+        msg = (
```

Comment:
~Can we put messages back into their exceptions, rather than defining an intermediate variable? (I could be missing context where it was decided this should happen, so I could be totally off base here).~

It looks like defining a `msg = ` variable is common in libraries like polars, so punting this comment 😅. If you know of a place where this approach is mentioned, might come from, I'd be curious to here -- since it'll be handy to point to!

Reply:
Oh, interesting -- I hadn't seen the error message lint rules before. I'm onboard with whatever seems most useful to folks. I wonder if these checks are disabled by default, and the select rule in pyproject.toml might have enabled them?

Opened issue #323 for tracking!